Cabal/src/Distribution/Simple/Utils.hs

```diff
@@ -374,6 +378,30 @@ die' verbosity msg = withFrozenCallStack $ do
     =<< pure . addErrorPrefix
     =<< prefixWithProgName msg
 
+-- Type which will be a wrapper for cabal -expections and cabal-install exceptions
+data VerboseException a = VerboseException CallStack POSIXTime Verbosity a
+  deriving (Show, Typeable)
+
+-- A new dieWithException function which will replace the existing die' call sites
+dieWithException :: HasCallStack => Verbosity -> CabalException -> IO a
+dieWithException verbosity exception = do
+  ts <- getPOSIXTime
+  throwIO $ VerboseException callStack ts verbosity exception
+
+-- Instance for Cabal Exception which will display error code and error message with callStack info
+instance Exception (VerboseException CabalException) where
+  displayException :: VerboseException CabalException -> [Char]
+  displayException (VerboseException stack timestamp verb cabalexception) =
+    withOutputMarker
+      verb
+      ( concat
+          [ "Error: [C-"
```

Comment:
Let's switch to "CABAL-", it is more informative. :)

Reply:
As per our discussion at the Tech Working Group, we should switch to "Cabal-"

Reply:
Sure, I have amended the changes, Its "Cabal-" now, Thanks